### PR TITLE
Many and IIngredient -> IData casters

### PR DESCRIPTION
**Pull request by Witixin1512**
This PR exposes the methods in the Many class to ZenCode, as well as adds IData casters for `IIngredientWithAmount` as well as `Many<KnownTag<ItemDefinition>>` to IData, using the `count` key. 

This allows users to use IData more seamlessly. This is intended to work in conjunction with `addJsonRecipe` scripts, using something like the following expansion:

```zenscript
import crafttweaker.api.data.IData;
import crafttweaker.api.data.ListData;
import crafttweaker.api.ingredient.IIngredient;

public expand IIngredient[] {
  public implicit as IData {
    var list = new ListData();
    for a in this {
      list.add(a as IData);
    }
    return list;
  }
}
```

This will be backported to 1.18.2 too.

### CODE REVIEW

File: Common/src/main/java/com/blamejared/crafttweaker/api/item/ExpandIIngredientWithAmount.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import com.blamejared.crafttweaker.api.tag.type.KnownTag;
+import com.blamejared.crafttweaker.api.util.Many;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import net.minecraft.world.item.Item;
+import org.openzen.zencode.java.ZenCodeType;
+
+@ZenRegister
+@Document("vanilla/api/item/ExpandIIngredientWithAmount")
+@ZenCodeType.Expansion("crafttweaker.api.ingredient.IIngredientWithAmount")
+public class ExpandIIngredientWithAmount {
```
Comment (jaredlll08) [inline Common/src/main/java/com/blamejared/crafttweaker/api/item/ExpandIIngredientWithAmount.java]: Why expand this compared to just having the method in the class as a default method?

File: Common/src/main/java/com/blamejared/crafttweaker/api/util/Many.java [hunk 2]
```diff
@@ -28,12 +28,14 @@ public Many(T data, int amount, Function<T, String> commandStringFunc) {
         this.commandStringFunc = commandStringFunc;
     }
     
-    
+    @ZenCodeType.Method
+    @ZenCodeType.Getter("data")
```
Comment (jaredlll08) [inline Common/src/main/java/com/blamejared/crafttweaker/api/util/Many.java:32]: Make sure that this getter actually works, I vaguely remember there being issues with generics and getters, but I don't remember if it applies to this case

Comment (Witixin1512) [reply Common/src/main/java/com/blamejared/crafttweaker/api/util/Many.java:32]: The getter works fine, tested it with the following:
```zenscript
val manySand = <tag:items:minecraft:sand> * 35;

println(manySand.data);
println(manySand.getData());
println(manySand.amount);
println(manySand.getAmount());
```

Which outputs:
```
<tag:items:minecraft:sand>
<tag:items:minecraft:sand>
35
35
```

File: Common/src/main/java/com/blamejared/crafttweaker/api/item/ExpandIIngredientWithAmount.java (hunk 1 above)
Comment (Witixin1512) [reply Common/src/main/java/com/blamejared/crafttweaker/api/item/ExpandIIngredientWithAmount.java]: Fixed.

**Review by jaredlll08 — APPROVED** [review]

**Pull request merged**